ci/scripts/swift_test.sh

```diff
@@ -22,9 +22,11 @@ set -ex
 data_gen_dir=${1}/swift/data-generator/swift-datagen
 export GOPATH=/
 pushd ${data_gen_dir}
-go get -d ./...
+git clone https://github.com/apache/arrow-go.git /arrow-go
```

Comment:
We could use a released tag but we can just clone main and find potential issues earlier. @kou any thoughts?

Reply:
How about doing this after we release a new version from apache/arrow-go?

I tried apache/arrow-go#87 for this but it didn't work.

Reply:
I wanted to remove `/go` subfolder before the Arrow release, if we do that we should release `arrow-go` before, was that your plan?

Reply:
Yes.

BTW, it seems that this script doesn't refer `go/` in apache/arrow. It uses released version: https://github.com/apache/arrow/blob/1c1f7f359d61b59e446b0a47e0279728551ed89f/swift/CDataWGo/go.mod#L21

This script may still work even if we remove `go/` in apache/arrow.